### PR TITLE
purescript-language-server 0.16.6 (new formula)

### PR DESCRIPTION
**Pull request by triallax**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/purescript-language-server.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require "language/node"
+
+class PurescriptLanguageServer < Formula
+  desc "Node-based Language Server Protocol server for PureScript"
```
Comment (mhmdanas) [inline Formula/purescript-language-server.rb:4]: Should this be made shorter?

```suggestion
  desc "Node-based LSP server for PureScript"
```

Maybe even "Node-based" could be removed?

File: Formula/purescript-language-server.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on "node"
+  depends_on "purescript"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "Connection input stream is not set.", shell_output("#{bin}/purescript-language-server 2>&1", 1)
```
Comment (mhmdanas) [inline Formula/purescript-language-server.rb:19]: We _can_ actually pretend to be an LSP client and communicate with the LSP server here, but I think this is sufficient.

File: Formula/purescript-language-server.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/purescript-language-server.rb:4]: yeah, `node-based` seems redundant, but `Language Server Protocol` is certainly better than `LSP`

File: Formula/purescript-language-server.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/purescript-language-server.rb:19]: You can also check with the other LSP tests so that some LSP JSON testing can be tested out.

Comment (mhmdanas) [reply Formula/purescript-language-server.rb]: I went ahead and did exactly that. Initially, I tried the `shell_output` approach (used in `bash-language-server` for instance), but the `purescript-language-server` invocation failed with exit code 1 for whatever reason. I then tried the `Open3.popen3` approach (used in `ansible-language-server`), and it worked just fine.

File: Formula/purescript-language-server.rb (hunk 1 above)
Comment (mhmdanas) [reply Formula/purescript-language-server.rb]: Done.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**